wifiutils/src/main/java/com/thanosfisherman/wifiutils/android10/Android10WifiConnectionReceiver.java

```diff
@@ -0,0 +1,176 @@
+        this.handler = new WeakHandler();
+    }
+
+    @Override
+    public void onReceive(final Context context, final Intent intent) {
+        final String action = intent.getAction();
+        wifiLog("Connection Broadcast action: " + action);
+
+        if (Objects.equals(WifiManager.SUPPLICANT_STATE_CHANGED_ACTION, action)) {
+            final SupplicantState state = intent.getParcelableExtra(WifiManager.EXTRA_NEW_STATE);
+            final int suppl_error = intent.getIntExtra(WifiManager.EXTRA_SUPPLICANT_ERROR, -1);
+            wifiLog("Connection Broadcast state: " + state);
+
+            if (state == null) {
+                return;
+            }
+
+            if (state == SupplicantState.DISCONNECTED && suppl_error == WifiManager.ERROR_AUTHENTICATING) {
+                wifiLog("Authentication error...");
+                handler.removeCallbacks(timeoutCallback);
+                cancelConnectingIfNeeded();
+                mWifiConnectionCallback.errorConnect(ConnectionErrorCode.AUTHENTICATION_ERROR_OCCURRED);
+            }
+        }
+    }
+
+    //region ConnectionHandler
+
+    @NonNull
+    @Override
+    public BroadcastReceiver connectWith(@NonNull ScanResult result, @NonNull final String password, @NonNull ConnectivityManager connectivityManager) {
+        mConnectivityManager = connectivityManager;
+
+        // schedule timeout behavior
+        handler.postDelayed(timeoutCallback, mDelay);
+
+        // start connection logic
+        connectWith(connectivityManager, result, password);
+
+        return this;
+    }
+
+    @Override
+    public void setTimeout(long millis) {
+        mDelay = millis;
+    }
+
+    //endregion
+
+    private void connectWith(@NonNull ConnectivityManager connectivityManager, @NonNull ScanResult scanResult, @NonNull final String password) {
+        WifiNetworkSpecifier.Builder wifiNetworkSpecifierBuilder = new WifiNetworkSpecifier.Builder()
+                .setSsid(scanResult.SSID)
+                .setBssid(MacAddress.fromString(scanResult.BSSID));
+
+        final String security = ConfigSecurities.getSecurity(scanResult);
+
+        ConfigSecurities.setupWifiNetworkSpecifierSecurities(wifiNetworkSpecifierBuilder, security, password);
+
+        final NetworkRequest networkRequest = new NetworkRequest.Builder()
+                .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+                .setNetworkSpecifier(wifiNetworkSpecifierBuilder.build())
+                .build();
+
+        // cleanup previous connections just in case
+        mDisconnectCallback.disconnect(connectivityManager);
+
+        mNetworkCallback = new ConnectivityManager.NetworkCallback() {
+            @Override
+            public void onAvailable(@NonNull Network network) {
+                super.onAvailable(network);
+
+                wifiLog("AndroidQ+ connected to wifi ");
+
+                // TODO: should this actually be in the success listener on WifiUtils?
```

Comment:
I'm not sure why you even moved this piece of code here in the first place. Shouldn't it belong to the `ConnectorUtils` as it used to?

Reply:
The problem I tried to solve this way was that ConnectorUtils doesn't have a way to asynchronously return results like how to broadcastreceivers can. In this callback: `new ConnectivityManager.NetworkCallback()` we can figure out if the user cancelled, if it was successful or something awful happened. Therefor I moved it to.

Reply:
I decided to move it back and change the WifiConnectionReceiver to handle Android10+ differently from previous versions.